src/classes/services/ContactService.cls

```diff
@@ -27,18 +26,36 @@ public with sharing class ContactService {
 
     public static void updateTeamMember(Map<Id,Contact> oldContactById, Map<Id,Contact> newContactById){
 
+        List<Contact> contactsToCreateNewTeamMember = new List<Contact>();
+
         Set<Id> contactIds = new Set<Id>();
 
-        for (Contact item : newContactById.values()){
-            if (item.Name != oldContactById.get(item.Id).Name || item.AccountId != oldContactById.get(item.Id).AccountId) {
-                contactIds.add(item.Id);
+        for (Contact item : oldContactById.values()){
+
+            if (String.isNotBlank(item.AccountId)) {
+
+                if (newContactById.get(item.Id).Name != item.Name || newContactById.get(item.Id).AccountId != item.AccountId) {
+
+                    contactIds.add(item.Id);
+                }
+
+            }else {
+
+                if (String.isNotBlank(newContactById.get(item.Id).AccountId)){
+
+                    contactsToCreateNewTeamMember.add(item);
+                }
             }
         }
 
+        createTeamMember(contactsToCreateNewTeamMember);
```

Comment:
MAJOR
After you gathered updated contacts there is no need to create new TM of Boutique Resource as I can see.
All you have to do is simply update existing ones:
```
        List<Boutique_Resource__c> boutiqueResources = BoutiqueResourceSelector.getBoutiqueResourcesByContactIds(updatedContactsIds);

        for (Boutique_Resource__c boutiqueResource : boutiqueResources) {
            boutiqueResource.Name = boutiqueResource.Contact__r.FirstName + LastName;
            boutiqueResource.Boutique__c = boutiqueResource.Contact__r.AccountId;
        }
        update boutiqueResources;
```